src/cpp/src/allocation_policy.cpp

```diff
@@ -0,0 +1,65 @@
+#include "coordinator.h"
+#include "communication.h"
+#include "message.h"
+#include <torch/torch.h>
+
+
+// Allocate partition
```

Comment:
Can we either use abstract methods or friend functions so that policies can coexist.

Reply:
Yes, we can use abstract methods, but need to define the interface properly before doing that which extends to multiple policies.